lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -0,0 +1,1890 @@
+            "type": "string",
+            "pattern": "^(2\\d\\d\\d)(:[2-9]\\d\\d\\d)?$",
+        },
+    },
+}
+DEFAULT_RELATION_NAME = "remote-configuration"
+DEFAULT_ALERTMANAGER_CONFIG_FILE_PATH = "/etc/alertmanager/alertmanager.yml"
+
+
+def load_config_file(path: str) -> dict:
+    """Reads given Alertmanager configuration file and turns it into a dictionary.
+
+    Args:
+        path: Path to the Alertmanager configuration file
+
+    Returns:
+        dict: Alertmanager configuration file in a form of a dictionary
+    """
+    if os.path.exists(path):
+        with open(path, "r") as config_yaml:
+            config = yaml.safe_load(config_yaml)
+        return config
+    else:
+        error_msg = "Given Alertmanager config file {} doesn't exist!".format(path)
+        logger.error(error_msg)
+        raise FileNotFoundError(error_msg)
+
+
+def load_templates_file(path: str) -> str:
+    """Reads given Alertmanager templates file and returns its content in a form of a string.
+
+    Args:
+        path: Alertmanager templates file path
+
+    Returns:
+        str: Alertmanager templates
+    """
+    if os.path.exists(path):
+        with open(path, "r") as template_file:
+            templates = template_file.read()
+        return templates
+    else:
+        logger.warning(
+            "Given Alertmanager templates file {} doesn't exist. Skipping...".format(path)
+        )
+        raise FileNotFoundError
+
+
+class AlertmanagerRemoteConfigurationProvider(Object):
+    """API that manages a provided `alertmanager_remote_configuration` relation.
+
+    The `AlertmanagerRemoteConfigurationProvider` is intended to be used by charms whose workloads
+    need to receive data from other charms' workloads over the `alertmanager_remote_configuration`
+    interface.
+
+    The `AlertmanagerRemoteConfigurationProvider` object can be instantiated as follows
+    in your charm:
+
+    ```
+    from charms.alertmanager_k8s.v0.alertmanager_remote_configuration import (
+        AlertmanagerRemoteConfigurationProvider,
+    )
+
+    def __init__(self, *args):
+        ...
+        self.remote_configuration_provider = AlertmanagerRemoteConfigurationProvider(self)
+        ...
+    ```
+
+    The `AlertmanagerRemoteConfigurationProvider` assumes that, in the `metadata.yaml`
+    of your charm, you declare a provided relation as follows:
+
+    ```
+    provides:
+        remote-configuration:  # Relation name
+            interface: alertmanager_remote_configuration  # Relation interface
+            limit: 1
+    ```
+
+    When `remote-configuration` is created, `AlertmanagerRemoteConfigurationProvider` reads
+    the current configuration of the Alertmanager from the specified configuration file and pushes
+    the content to the relation data bag. This initial configuration can then be used
```

Comment:
The `alertmanager-configurer` can't generate entire config file at once. In fact, it can't generate the config file at all. The absolute minimum it requires to work properly is an empty file representing Alertmanager config.
I considered 2 alternative options:

1. Using an empty file
2. Using a dummy default config file delivered with the `alertmanager-configurer`

Problem with using an empty file is that creating a complete Alertmanager config requires at least a couple of REST calls. Each change of the config file however causes updating the relation data bag. This means Alertmanager would receive couple of incomplete or invalid configs by the time `alertmanager-configurer` would prepare the final config.
Problem with using a default config delivered along with the `alertmanager-configurer` is similar as above. 
Additional problem with both above approaches is that they would overwrite the existing config of Alertmanager. 
Loading current Alertmanager config upon relation creation solves all mentioned problems. 
Additionally, if someone decides not to use `alertmanager-configurer-k8s` but to develop own custom configurer, he doesn't have use the initial config.

Another alternative that I just though of, would be exposing Alertmanager API url in the relation data so that any `configurer` charm could fetch the current configuration of Alertmanager using the `status` endpoint...

Reply:
As we discussed, this has been abandoned in the new implementation.